### PR TITLE
Handle linking to unregistered hosts

### PR DESCRIPTION
**Pull request by rtorrero**
Captura desde 2023-06-21 09-50-03

# Description

Check for unregistered hosts to avoid linking and show a warning

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
For the record, knowing that the PR is in draft.
This change will cause massive conflicts on main, as this file is even deleted.
I encourage to base this work on `main` branch.

**Review by arbulu89 — COMMENTED** [review]
@rtorrero I think we could put this logic in the already existing `HostLink` component, or create a new one, as we need to have this logic in the ascs/ers clusters as well.
What do you think?

**Comment by rtorrero** [comment]
> @rtorrero I think we could put this logic in the already existing `HostLink` component, or create a new one, as we need to have this logic in the ascs/ers clusters as well. What do you think?

I had this conversation a few weeks with @dottorblaster and we went for the current approach to avoid adding artifacts to places where `HostLink` might have been used in a different context.

But your point avoid avoiding duplicating the logic in the ascs/ers clusters is a valid concern. As you suggested in Slack, creating a separate `ClusterNodeLink` component seems the way to go.

File: assets/js/components/ClusterDetails/AscsErsClusterDetails.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/AscsErsClusterDetails.test.jsx:204]: This is already tested in `should show nodes information`.
It has a typo though.
In line 108 we should jave `id: hostID` instead of `id: clusterID`.
If you want to have an alternative test, we should test that when the host is not there, we don't have a link

File: assets/js/components/ClusterDetails/HanaClusterDetails.stories.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/HanaClusterDetails.stories.jsx:86]: I would use the `slice` even here

File: assets/js/components/ClusterDetails/HanaClusterDetails.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/HanaClusterDetails.test.jsx:188]: Do we need this fixes name?
The factory will already create a name. `registeredClusterNode.name`

File: assets/js/components/ClusterDetails/ClusterNodeLink.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterNodeLink.jsx:7]: I would use the same strategy as `HostLink` and pass the name as `children`,
so you can use `<ClusterNodeLink hostId={id}>name</ClusterNodeLink>`

File: assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx:9]: Let's use `HostFactory`

File: assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx:25]: Factory

File: assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterNodeLink.test.jsx:14]: Do you need this `path` and `route` thing?
You are not clicking the url.
I think `renderWithRouter` would be enough

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Looking good!
Some minor changes and we are ready to merge
@jagabomb could you have a look on how it looks?
Changes available in PR storybook. Stories `AscsErsClusterDetails` and `HanaClusterDetails`

**Review by arbulu89 — APPROVED** [review]
Much better :smiley: 
Feel free to merge once you have @jagabomb 's approval

**Review by CDimonaco — APPROVED** [review]
Good job LGTM when @jagabomb approves

**Review by jagabomb — APPROVED** [review]
LGTM. I would like to also double check this with @abravosuse  when he is back. But so far this solution makes sense to me.

**Review by nelsonkopliku — APPROVED** [review]

**Review by abravosuse — APPROVED** [review]
I have reviewed AscsErsClusterDetails With Unregistered Host and HanaClusterDetails With Unregistered Host in the Storybook and they both look good to me.

**Pull request merged**
